exchange/utils.go

```diff
@@ -263,6 +265,39 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 	return
 }
 
+// fpdUserEIDExists determines if req fpd config had User.EIDs
+func fpdUserEIDExists(req *openrtb_ext.RequestWrapper, fpd map[openrtb_ext.BidderName]*firstpartydata.ResolvedFirstPartyData, bidder string) bool {
+	fpdToApply, exists := fpd[openrtb_ext.BidderName(bidder)]
+	if !exists || fpdToApply == nil {
+		return false
+	}
+	if fpdToApply.User == nil {
+		return false
+	}
+	fpdUserEIDs := fpdToApply.User.EIDs
+	if len(fpdUserEIDs) == 0 {
+		return false
+	}
+
+	reqUserEIDs := req.User.EIDs
+
+	if len(reqUserEIDs) != len(fpdUserEIDs) {
+		return true
+	}
+
+	// if bidder fpd didn't have user.eids then user.eids will remain the same
+	// hence we can use the same index to compare elements
+	for i := range reqUserEIDs {
+		pReqUserEID := &reqUserEIDs[i]
+		pFpdUserEID := &fpdUserEIDs[i]
+		if pReqUserEID != pFpdUserEID {
+			return false
+		}
+
+	}
+	return true
```

Comment:
Isn't this supposed to be `return false`?

Reply:
Refactored